### ISSUE TITLE
Custom variants of toast when using headless mode

### ISSUE THREAD

**Issue by mrkldshv** [timolins/react-hot-toast#23]
Hi! Is it possible to provide any custom variants of toast with the headless mode? It would be great, if we could create `Notifications` component and have own `success`, `error` and other toast's variants to use it, as: `toast.custom('My custom toast variant')`.

**Comment by mverissimo** [comment]
I think that you can do something like this:

```
import { toast as handler } from 'react-hot-toast'

toast.info = msg => handler(msg, {
    style: {}
});

export { toast }
```

**Comment by mrkldshv** [comment]
@mverissimo I'd like to achieve slightly different result. Using `useToaster` hook, I'd create my own notifications with headless mode and have custom toast variants. [Documentation](https://react-hot-toast.com/docs/use-toaster).

**Comment by timolins** [comment]
Thanks for the input. This definitely something that should be possible, but isn't right now.

Would an API like this work for you? 


```jsx

import { createCustomToaster } from 'react-hot-toast'

const { useToaster, Toaster, toast } = createCustomToaster([ "info", "custom" ])

<Toaster
  toastOptions={{
    info: {
     icon: "💡"
    }
  }}
/>

toast.info("Did you know...")

```

**Comment by mrkldshv** [comment]
Yep, this looks good to me! It would be great if I can just pass custom toast variant to the `Toaster` as in your example, without creating completely custom component with headless mode.

**Comment by timolins** [comment]
The benefit of the proposed method is, that it also adds the toast.custom() to the toast function. Also it makes all options type-safe with auto-complete.

**Comment by mrkldshv** [comment]
Yeah, exactly. I really like the method you proposed.

**Comment by lsbyerley** [comment]
@ZachHaber do you mind expanding on how you setup a custom `Alert` toast? I am looking to add one as well

**Comment by lsbyerley** [comment]
@ZachHaber thanks for the detailed example! I think I may try to hold out for v2 and just style the default toast as an alert for now. Not sure when v2 is planned to release though

**Comment by MHase** [comment]
hey guys 👋🏻 
any progress on that feature or maybe `simple` workaround? I have similar issue for `warning` variant and it doesn't seem too intuitive to use e.g. `toast('This is warning message');` by default

**Comment by char502** [comment]
Hi, has there been any progress on this, it would be really useful just to be able to create a warning variant type and similar?
